nnvm/src/top/tensor/transform.cc

```diff
@@ -132,11 +132,9 @@ inline bool ConcatenateCorrectLayout(const NodeAttrs& attrs,
   CHECK_EQ(olayouts->size(), 1U);
 
   for (size_t i = 0; i < ilayouts->size(); ++i) {
-    const Layout& input = last_ilayouts->at(i).defined() ?
-                          last_ilayouts->at(i) : ilayouts->at(i);
-    NNVM_ASSIGN_LAYOUT(*ilayouts, i, input);
+    NNVM_ASSIGN_LAYOUT(*ilayouts, i, ilayouts->at(0));
```

Comment:
need to first check `param.axis` <= current layout size. otherwise it must be transformed to original layout first.